packages/components/src/components/hds/stepper/nav/index.ts

```diff
@@ -37,8 +36,11 @@ export interface HdsStepperNavSignature {
     body?: [];
     default: [
       {
-        Step?: ComponentLike<HdsStepperNavStepSignature>;
-        Panel?: ComponentLike<HdsStepperNavPanelSignature>;
+        Step?: WithBoundArgs<typeof HdsStepperNavStepComponent, 'currentStep'>;
+        Panel?: WithBoundArgs<
+          typeof HdsStepperNavPanelComponent,
+          'currentStep'
```

Comment:
AFAIK we have to add all the arguments that are passed in the `yield` declaration: https://github.com/hashicorp/design-system/blob/8066b466aec268b932c6636b17f42da9e1fd7da6/packages/components/src/components/hds/stepper/nav/index.hbs#L29-L37

(if unsure, maybe check with @aklkv, he's the one who told me this thing)